src/sentry/db/models/outboxes.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Collection, List, Mapping, Type
+
+from sentry.types.region import find_regions_for_orgs
+
+if TYPE_CHECKING:
+    from sentry.models.outbox import ControlOutboxBase, OutboxCategory, RegionOutboxBase
+
+
+class ProducesRegionOutboxOnUpdate:
```

Comment:
I'll add more documentation strings heres.

The idea is that we have two heirarchies, one for region and one for contorl models, in two layers

1.  ProducesOutboxOnUpdate layer, which is indifferent to which outboxes a save/update/delete fires.
2. ProcessUpdatesWith layer, which given some class level variables knows how to "generally" produce, fire, and process save/update/delete outboxes given some base line assumptions that can be configured.

Reply:
There isn't much in this layer right now. Are there plans to expand it in the future? Right now I would probably merge this with `ProcessUpdatesWithRegionOutboxes` as this class isn't doing enough to justify keeping it around.

Reply:
`Produces*` heirarchy is the signal for the `save/update/delete` overrides that write outboxes, but NOT the signal for hooking up a default receiver.  The distinction seemed useful in the special case that you may want to manually wire up the receivers rather than use the model machinery as is.  I haven't yet demonstrated a need for it, but I have a hunch that such special cases will exist.  Thoughts?